packages/mobile/src/send/utils.ts

```diff
@@ -0,0 +1,269 @@
+  let recipientAddress: string | null | undefined
+
+  if (recipient.kind === RecipientKind.QrCode || recipient.kind === RecipientKind.Address) {
+    recipientAddress = recipient.address
+  } else if (recipient.e164PhoneNumber) {
+    recipientAddress = getAddressFromPhoneNumber(
+      recipient.e164PhoneNumber,
+      e164NumberToAddress,
+      secureSendPhoneNumberMapping
+    )
+  }
+
+  return { ...transactionData, recipientAddress }
+}
+
+export const getVerificationStatus = (
+  recipient: Recipient,
+  e164NumberToAddress: E164NumberToAddressType
+): RecipientVerificationStatus => {
+  if (recipient.kind === RecipientKind.QrCode || recipient.kind === RecipientKind.Address) {
+    return RecipientVerificationStatus.VERIFIED
+  }
+
+  if (recipient.e164PhoneNumber) {
+    const addresses = e164NumberToAddress[recipient.e164PhoneNumber]
+
+    if (addresses === undefined) {
+      return RecipientVerificationStatus.UNKNOWN
+    }
+
+    if (addresses === null) {
+      return RecipientVerificationStatus.UNVERIFIED
+    }
+
+    return RecipientVerificationStatus.VERIFIED
+  }
+
+  return RecipientVerificationStatus.UNKNOWN
+}
+
+export const getFeeType = (
+  recipientVerificationStatus: RecipientVerificationStatus
+): FeeType | null => {
+  switch (recipientVerificationStatus) {
+    case RecipientVerificationStatus.UNKNOWN:
+      return null
+    case RecipientVerificationStatus.UNVERIFIED:
+      return FeeType.INVITE
+    case RecipientVerificationStatus.VERIFIED:
+      return FeeType.SEND
+  }
+}
+
+// Given addresses can be added and deleted we can't rely on changes in length to signal address changes
+// Not sure if address order in array is consistent so not assuming it
+const newAddressesAdded = (oldAddresses: string[], newAddresses: string[]) => {
+  const oldAddressesSorted = oldAddresses.sort()
+  const newAddressesSorted = newAddresses.sort()
+
+  for (let i = 0; i < newAddressesSorted.length; i += 1) {
+    if (oldAddressesSorted[i] !== newAddressesSorted[i]) {
+      return true
+    }
+  }
+
+  return false
+}
+
+const last4DigitsAreUnique = (addressArr: string[]) => {
+  const last4DigitArr = addressArr.map((address) => address.slice(-4).toLowerCase())
+  const last4DigitSet = new Set()
+  last4DigitArr.forEach((endDigits) => last4DigitSet.add(endDigits))
+  return last4DigitArr.length === last4DigitSet.size
+}
+
+// Check if there are multiple addresses but somehow a preferred addresses hasn't been set yet
+// Should never happen in production but makes the change backwards compatible for testing env
+const accidentallyBypassedValidation = (
+  newAddresses: string[] | null,
+  e164Number: string,
+  secureSendPhoneNumberMapping: SecureSendPhoneNumberMapping
+) => {
+  const validatedAddress =
+    secureSendPhoneNumberMapping[e164Number] || secureSendPhoneNumberMapping[e164Number].address
+  return newAddresses && newAddresses.length > 1 && !validatedAddress
+}
+
+export const checkIfValidationRequired = (
```

Comment:
How about we move all this stuff to `identity/secureSend.ts`?

Reply:
Sounds good